pandas/core/dtypes/dtypes.py

```diff
@@ -817,6 +848,10 @@ def __eq__(self, other):
 
         return isinstance(other, PeriodDtype) and self.freq == other.freq
 
+    def __setstate__(self, state):
+        # for pickle compat.
```

Comment:
why exactly is this needed? e.g. what breaks if you don't have it

Reply:
Pickling, as it is saving `freq` in `__getstate__` (see eg failures in https://travis-ci.org/pandas-dev/pandas/jobs/520153338)